### DISCUSSION TITLE
Atoum error extends class

### DISCUSSION THREAD

**Discussion by kzacristo** [atoum/atoum#896] [Q&A] [answered]
I am implementing unit tests in my GLPI system using atoum, installed via composer, was created a boostrap file.php as the documentation sends and created the first test case but when the class extend another class it presents an error in the images below this error, folder structure and the configuration file and the test case

Error.
![Captura de tela 2023-08-07 111111](https://github.com/atoum/atoum/assets/13594787/13008e02-8dcf-4c28-8dd1-e47485e0ddf0)

path
![Captura de tela 2023-08-07 111244](https://github.com/atoum/atoum/assets/13594787/6f632cd1-3125-4bc0-85d6-5a1f72f1a47b)

coding
![Captura de tela 2023-08-07 111551](https://github.com/atoum/atoum/assets/13594787/2731950a-bb05-4a0a-a787-f5dd8dbe3fac)

**Comment by Grummfy** [comment, marked as answer]
hello,
So I will try to help you following what I see, notice that I don't know GLPI so perhaps there is element that can interract with.
But
* there should have no reason to include the bootstrap file, inside the test file. the fact that you provide the bootstrap file in the cli, should be enough
* I don't really understand why you initialized a plugins in the bootstrap file that you seems to test after
* your error seems related to a file you try to include inside the boostrap file => the file seems to not exist in the list of file you provide


I would recommend to read theses documentations that could help
* https://atoum.readthedocs.io/en/latest/first_test.html
* https://atoum.readthedocs.io/en/latest/mode-debug.html

Answer is a bit short, didn't got many time to dig deeper

**Reply by kzacristo** [reply]
In the GLPI documentation it is asked to create a file to test the plugin you want and should be placed next to the plugin to be tested link of the documentation https://glpi-developer-documentation.readthedocs.io/en/master/plugins/test.html
